models/signature.js

```diff
@@ -78,6 +61,75 @@ function parseSignatures(data) {
    return signatures;
 }
 
+/**
+ * Parses a GitHub review and returns an array of Signature objects.
+ * Returns an empty array if the comment did not contain any of our tags.
+ */
+Signature.parseReview = function parseReview(review, repoFullName, pullNumber) {
+   const reviewData = {
+      repo: repoFullName,
+      number: pullNumber,
+      body: review.body,
+      user: review.user,
+      created_at: review.submitted_at,
+      id: review.id,
+      state: review.state
+   }
+
+   return parseReviewSignature(reviewData);
+};
+
+function parseReviewSignature(data) {
+   let signatures = [];
+
+   if (data.state === 'APPROVED') {
+      signatures.push(new Signature({
+         repo: data.repo,
+         number: data.number,
+         user: {
+            id:    getUserid(data.user),
+            login: getLogin(data.user)
+         },
+         type: 'CR',
+         created_at: data.created_at,
+         active: true,
+         comment_id: data.id
+      }));
+   }
+
+   if (data.state === 'DISMISSED' || data.state === 'PENDING') {
+      signatures.push(new Signature({
```

Comment:
I'm tempted to say that we should create the object up top and only mutate the `type` and `active` in the conditions. Not for code reduction, but to reduce the risk of a change updating one of these ifs and forgetting the other.

Thoughts? This is minor.

Reply:
The problem with this is that there is also a `COMMENT` status. We don't want to do anything on that I don't think. I could just bail early if it's a `COMMENT`.

Reply:
I see.

![image](https://user-images.githubusercontent.com/26855/191135762-b0a44d05-2f9c-4fcc-97b1-610924359207.png)

Should we interpret comments as anything? Maybe deploy block? i.e. someone is saying "I've looked at this and I have questions I want addressed before I approve, but I don't have specific changes I want".

I feel like many of my CRs end up in that state. Thoughts?

Reply:
If the type of the review is not one that we care about (i.e. `COMMENT`), we ignore it now.

Reply:
Sometimes I just comment `"Nice!"` or something like that on the comment. I think we just ignore it.